pkg/analyzer/lib/src/dart/analysis/file_state.dart

```diff
@@ -716,7 +716,7 @@ class FileSystemState {
 
       // If the URI cannot be resolved, for example because the factory
       // does not understand the scheme, return the unresolved file instance.
-      if (uriSource == null) {
+      if (uriSource == null || !uriSource.exists()) {
```

Comment:
I don't understand the need for this change.
It seems wrong to me.

The URI is either can be resolved and is always resolved to this file (until the `SourceFactory`, `FileSystemState` and `AnalysisDriver` recreation), or it cannot be resolved at all and `SourceFactory` returns `null`. It's OK for the file to not exist, we handle this OK in `FileState.refresh()` and use the empty string as its content. For Bazel if there is no corresponding generated file, we default the URI resolution to the corresponding file in the `lib` folder of the corresponding package.